### DISCUSSION TITLE
Are shortcodes within html comments <!-- always --> processed?

### DISCUSSION THREAD

**Discussion by ute** [quarto-dev/quarto-cli#8912] [Q&A] [answered]
### Description

I stumbled upon a behavior that I would not expect:  
html comments do not preclude shortcodes from being read and throwing an error.

```markdown
Include shortcut within comment
<!-- 
{{< include notfound.qmd >}}
-->
is apparently processed anyway.
```

An error is thrown if file `notfound.qmd` cannot be found.  
I am wondering if this is intended? it could be used to switch comments on and off, e.g. if `notfound.qmd` consists of

```markdown
-->

> And the comment is turned off, and on again

<!--
hidden stuff
```

**Comment by mcanouil** [comment, marked as answer]
Yes, because shortcode are evaluated by Quarto while `<!-- ` is a Pandoc feature so resolves at rendering not at pre-processing.
If you don't want your shortcode to be processed, you need to escape them `{{{< include notfound.qmd >}}}`.

See:
- https://github.com/quarto-dev/quarto-cli/issues/5100
- https://github.com/quarto-dev/quarto-cli/discussions/3330
